### PR TITLE
Add docs for useTagGroup

### PR DESCRIPTION
**Pull request by reidbarber**
Note: Merge and update styled example after release

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Test useTagGroup docs page and styled example.

### CODE REVIEW

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (snowystinger) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:38]: we should include useTag here too? like the useMenu docs
```
<FunctionAPI function={docs.exports.useTag} links={docs.links} />
```

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (snowystinger) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:50]: I'm sure you're already handling this, just making a note, anatomy should include row/cell like gridlist diagram

**Review by snowystinger — COMMENTED** [review]
looks pretty good, just small comments

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (reidbarber) [reply packages/@react-aria/tag/docs/useTagGroup.mdx:51]: Added!

**Review by snowystinger — DISMISSED** [review]
will need to wait for next release and styled example will need updating then as well. otherwise, looks good

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (ktabors) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:137]: I noticed that `useTagGroup` above lists the return props in the same order as they are listed in the `Anatomy` section. Here `clearButtonProps` of `useTag` is first instead of last, like it would be if it mirrored the order from the `Anatomy` section.

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (ktabors) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:235]: In this codesandbox Tailwind example and the Remove tags example below, you can shift-tab and get to the remove/clear button of a Tag. Is that a feature or bug?

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (ktabors) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:245]: Space key works to remove the tag too, but isn't mentioned here, should it be?

**Review by ktabors — COMMENTED** [review]
Looks good, had a couple questions.

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (ktabors) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:288]: Discussed with Reid and the second sentence doesn't make sense for this hook.

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (reidbarber) [reply packages/@react-aria/tag/docs/useTagGroup.mdx:245]: This is a good point, although outside of this docs PR.

I could see the argument that we keep it to Delete and Backspace right now, so that space is reserved for selection.

However, I don't know if selection and removal will be supported at the same time.

I can bring this up with the team.

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (snowystinger) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:277]: I feel like 'Your selected categories' would be better. There is nothing interactive where you can perform a selection right now, so this wording is a bit odd. Same thing in the error message case. Maybe 'Invalid set of categories'? just to reinforce that we didn't do a selection?

**Review by snowystinger — COMMENTED** [review]
nit

**Review by snowystinger — APPROVED** [review]

**Comment by reidbarber** [comment]
> we've updated the codesandbox to point to the latest?

@snowystinger there are some prop name changes that have been merged, so I'll update those in the sandbox after release and before this is merged. I suppose I could also just use nightly in the sandbox for now.

**Review by snowystinger — APPROVED** [review]

**Review by ktabors — APPROVED** [review]

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (devongovett) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:24]: Noticed that we put this in the Status category for Spectrum. Which one makes more sense? I kinda like collections but not sure

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (devongovett) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:30]: IMO this description is a bit weird, maybe too spectrum-y? "Tags allow users to categorize content. They can represent keywords or people, and are grouped to describe an item or a search request."

Maybe "A tag group is a focusable list of labels, categories, keywords, or other items, with support for keyboard navigation and removal."

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (devongovett) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:52]: Maybe we should also include the tag group label in this diagram? Might be confusing since there is something called "Label" there but it's the individual tag label rather than the one for the whole tag group. Maybe make that more specific too.

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (devongovett) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:151]: I guess the button should be inside the grid cell? Might have to break the label out into a separate element and nest the both inside the cell.

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (reidbarber) [reply packages/@react-aria/tag/docs/useTagGroup.mdx:24]: I'll move into Status for now top match Spectrum. We can revisit this before release or later since it's non-breaking.

File: packages/@react-aria/tag/docs/useTagGroup.mdx
Comment (devongovett) [inline packages/@react-aria/tag/docs/useTagGroup.mdx:138]: Just noticed this. Why does useTag need an `isFocused` prop? Seems like it also computes that itself internally.

**Review by snowystinger — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**
